### ISSUE TITLE
404 Ajax Request for Chooser Widgets

### ISSUE THREAD

**Issue by dfelton** [tzyganu/UMC1.9#96]
The code generated in the `Namespace_Module_Block_Adminhtml_Entity_Widget_Chooser::prepareElementHtml` method (replacing "Namespace", "Module", and "Entity" in the class name for actual values) references the frontend URL to access the WidgetController used in the AJAX request. Because the controller class extends Mage_Adminhtml_Controller_Action, accessing it using the frontend URL to it results in a 404 response. 

**Example Source Generated:**
```php
    $sourceUrl = $this->getUrl(
        'northwoods_pulls/adminhtml_pulls_salecategory_widget/chooser',
        array('uniq_id' => $uniqId)
    );
```

**What Should Be Generated:**
```php
    $sourceUrl = $this->getUrl(
        '*/pulls_salecategory_widget/chooser',
        array('uniq_id' => $uniqId)
    );
```

**Comment by dfelton** [comment]
Update: This bug only affects entities which are configured to NOT behave like a tree. 

The bug lies in this file:
`app/code/community/Ultimate/ModuleCreator/etc/source/app/code/Block/Adminhtml/Entity/Widget/Chooser/020_not_tree`

**Line 32:** `'{{namespace}}_{{module}}/adminhtml_{{module}}_{{entity}}_widget/chooser',`
**Update To:** `'*/{{module}}_{{entity}}_widget/chooser',`

Thanks for the extension! Total life safer for streamlining common tasks at the start of any project.

**Comment by dfelton** [comment]
Last note: This bug exists in the version I installed via MagentoConnect (1.9.5.0), not GitHub. Not sure how religiously you make sure the MagentoConnect version is the latest from GitHub, so thought I'd point this out.

**Comment by tzyganu** [comment]
@dfelton Thanks for spotting this. 
I admit, I kind of neglected the version on MagentoConnect. I don't push every new version that I push on github.  
Also I don't understand what you mean that the bug does not exist in the version from github. I can clearly see the wrong line [in here](https://github.com/tzyganu/UMC1.9/blob/master/app/code/community/Ultimate/ModuleCreator/etc/source/app/code/Block/Adminhtml/Entity/Widget/Chooser/020_not_tree#L31).  
You are right, it should be `'*/{{module}}_{{entity}}_widget/chooser'` instead of `'{{namespace}}_{{module}}/adminhtml_{{module}}_{{entity}}_widget/chooser'`.  
I will fix this as soon as I have some time.  
It could even go faster if you make a pull request for it.  
Thank you again fo making the UMC better.

**Comment by dfelton** [comment]
@tzyganu Thanks for the reply. 

Sorry I didn't mean "the bug does not exist in the version from GitHub", I merely meant that I did not check the GitHub version myself, I had only looked at the version available on MagentoConnect.

My GitHub skills are close to none, however this may be a good excuse to finally familiar myself with Git. When I have time myself I may contribute what I can. 

Have a great day.
